daprdocs/content/en/reference/components-reference/supported-middleware/middleware-routerchecker.md

```diff
@@ -0,0 +1,71 @@
+---
+type: docs
+title: "RouterChecker http request routing"
+linkTitle: "RouterChecker"
+description: "Use routerchecker middleware to block invalid http request routing"
+aliases:
+- /developing-applications/middleware/supported-middleware/middleware-routerchecker/
+---
+
+The RouterChecker HTTP [middleware]({{< ref middleware.md >}}) component leverages regexp to check the validity of HTTP request routing to prevent invalid routers from entering the Dapr cluster. In turn, filtering out bad requests reduces noise in the telemetry and logs data.
+
+## Component format
+
+In the following definition,  the HTTP request RouterChecker is set to validate against `^[A-Za-z0-9/._-]+$`:
+
+```yaml
+apiVersion: dapr.io/v1alpha1
+kind: Component
+metadata:
+  name: routerchecker 
+spec:
+  type: middleware.http.routerchecker
+  version: v1
+  metadata:
+  - name: rule
+    value: "^[A-Za-z0-9/._-]+$"
+```
+
+As an example, the above definition would result in the following PASS/FAIL cases:
+
+```shell
+PASS /v1.0/invoke/demo/method/method
+PASS /v1.0/invoke/demo.default/method/method
+PASS /v1.0/invoke/demo.default/method/01
+PASS /v1.0/invoke/demo.default/method/METHOD
+PASS /v1.0/invoke/demo.default/method/user/info
+PASS /v1.0/invoke/demo.default/method/user_info
+PASS /v1.0/invoke/demo.default/method/user-info
+
+FAIL /v1.0/invoke/demo.default/method/cat password
+FAIL /v1.0/invoke/demo.default/method/" AND 4210=4210 limit 1
+FAIL /v1.0/invoke/demo.default/method/"$(curl
+```
+
+## Spec metadata fields
+
+| Field | Details | Example |
+|-------|---------|---------|
+| rule | the regexp expression to be used by the HTTP request RouterChecker | `^[A-Za-z0-9/._-]+$`|
```

Comment:
Can you be clear here on the regex rules. Can I only have this `^[A-Za-z0-9/._-]+$` or can I have ANY regex? It would be good to be clear on what regex expressions are valid.

Reply:
I checked the code changes and found that any regex can be supported.
@msfussell